mcs/class/System.Private.CoreLib/System.Threading/Mutex.cs

```diff
@@ -1,14 +1,80 @@
+using System.Runtime.CompilerServices;
+using System.IO;
+
 namespace System.Threading
 {
 	partial class Mutex
 	{
-		void CreateMutexCore (bool initiallyOwned, string name, out bool createdNew) => throw new NotImplementedException ();
-
-		static OpenExistingResult OpenExistingWorker (string name, out Mutex result) => throw new NotImplementedException ();
+		Mutex (IntPtr handle) => Handle = handle;
 
 		public void ReleaseMutex ()
 		{
-			throw new NotImplementedException ();
+			if (!ReleaseMutex_internal (Handle))
+				throw new ApplicationException ("Mutex is not owned");
+		}
+
+		void CreateMutexCore (bool initiallyOwned, string name, out bool createdNew) =>
+			Handle = CreateMutex_internal (initiallyOwned, name, out createdNew);
+
+		unsafe static IntPtr CreateMutex_internal (bool initiallyOwned, string name, out bool created)
+		{
+			fixed (char *fixed_name = name)
+				return CreateMutex_icall (initiallyOwned, fixed_name,
+					name?.Length ?? 0, out created);
+		}
+
+		static OpenExistingResult OpenExistingWorker (string name, out Mutex result) 
+		{
+			if (name == null)
+				throw new ArgumentNullException (nameof (name));
+
+			result = null;
+			if ((name.Length == 0) ||
+				(name.Length > 260)) {
+				return OpenExistingResult.NameInvalid;
+			}
+			
+			MonoIOError error;
+			IntPtr handle = OpenMutex_internal (name, MutexRights.Modify, out error);
```

Comment:
use `out MonoIOError error` instead

Reply:
These are probably the same icalls used by the normal version.